main.rb

```diff
@@ -0,0 +1,108 @@
+require_relative 'bank_account'
+require_relative 'bank_system'
+
+# Main program
+bank_system = BankSystem.new
+
+puts "Welcome! \n What do you want to do?"
+
+loop do
+  puts "\n Press: \n 1 - create account; \n 2 - display all the accounts; \n 3 - retrieve money; \n 4 - deposit money; \n 5 - display current balance; \n 6 - exit. "
+
+  choice = gets.chomp.to_i
+
+  case choice
+  when 1
+    puts "Hi! \n Insert your name."
+    name = gets.chomp
+    name.downcase!
+    name.capitalize!
+
+    puts "#{name}, insert your job."
+    job = gets.chomp
+    job.downcase!
+
+    puts "Insert your email."
+    email = gets.chomp
+    email.downcase!
+
+    puts "Insert your address."
+    address = gets.chomp
+    address.downcase!
+
+    account = bank_system.create_account(name, job, email, address)
+
+    puts "Set a PIN for your account. \n It needs to be 4 digits long. "
+    pin = gets.chomp
+
+    while (pin.length > 4 || pin.length < 4)
+      puts "The chosen PIN is not valid. \n A 4-digit PIN is needed. "
+      pin = gets.chomp
+    end
+
+    account.set_pin(pin)
+  when 2
+    bank_system.display_accounts
+  when 3
+    puts "\n Insert your name. "
+    name = gets.chomp
+    account = bank_system.find_account_by_name(name)
+    if account.nil?
```

Comment:
I noticed there are multiple places where you are using
```ruby
if amount.nil?
  puts "error message"
end
```
or other similar conditions like `amount <= 0`

I think the best option for this would be something like an error raise. You can also avoid the error raise in case you would want the program to continue (an error raise would need handling to not stop it).

Or a second option, you could extract logic to validate amount into a method that can be easily reused. You will always check for the same things